app/lib/three_scale/domain.rb

```diff
@@ -1,19 +1,20 @@
 module ThreeScale
   module Domain
+    STANDARD_HTTP_PORTS = [80, 443].freeze
+
     def self.current_endpoint(request, host = request.host)
-      host_and_port = if Rails.env.development?
-                        request.host_with_port("#{host}:#{request.port}")
-                      elsif Rails.env.preview?
-                        request.respond_to?(:real_host) ? request.real_host(host) : host
-                      else
-                        host
-                      end
-      "#{request.scheme}://#{host_and_port}"
+      scheme = request.scheme
+      port = request.port.to_i
+      if STANDARD_HTTP_PORTS.exclude?(port)
+        "#{scheme}://#{host}:#{port}"
+      else
+        "#{scheme}://#{host}"
+      end
     end
 
     def self.callback_endpoint(request, account, host = request.host)
       endpoint = if account.master?
-                   current_endpoint(request, account.domain) + '/master/devportal'.freeze
+                   current_endpoint(request, account.external_domain) + '/master/devportal'.freeze
```

Comment:
I know that what I am gonna say is not related to your PR but... does this work? 😕 
First of all, `master` does not have a developer portal so if this is correct, the name is definitely confusing. Second, and more importantly, when I check for this route I find only this:
![image](https://user-images.githubusercontent.com/11318903/89029449-c37e1d80-d32e-11ea-95b7-af3368136860.png)

Reply:
Master domain == self_domain 
It will work for all providers

Reply:
I understood that part @hallelujah . The part that concerns me is why it is `'/master/devportal'` and not `'/master/devportal/auth'`. And "what is this 'devportal' thing of master?".

This is why I said that it is not related to your PR. If it was working before your PR, it will keep working now, and if it wasn't working before your PR, then for sure it does not work now yet xD

But if you say that it works, I trust you.

Reply:
Because it is just a method, but does not return the whole path ... used somewhere else